File: src/Java/3.LengthOfLongestSubstring.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;
import java.util.HashSet;


class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();
        if (n < 2) return n;
        int[] record = new int[128];
        int left = 0;
        int res = 0;
        for (int right = 0; right < n; right++){
            char cur = s.charAt(right);
            if (record[cur] > left){
                left = record[cur];
            }
            record[cur] = right + 1;     
            res = Math.max(res, right - left + 1);
        }     
        return res;
    }
}

File: src/CSharp/3.LengthOfLongestSubstring.cs

using System;
using System.Diagnostics.Contracts;

namespace PeefyLeetCode.LengthOfLongestSubstring
{
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int[] hax = new int[128];
        Array.Fill(hax, -1);
        int res = 0;
        int left = -1;
        for(int i = 0; i < s.Length; ++i){
            left = Math.Max(left, hax[s[i]]);
            hax[s[i]] = i;
            res = Math.Max(i - left, res);
        }
        return res;
    }
}
}